torch/nn/modules/module.py

```diff
@@ -20,6 +20,20 @@ def _addindent(s_, numSpaces):
     return s
 
 
+def _first_var(input, kwargs):
```

Comment:
I'm not sure why this is necessary / why it works at all, because from where I'm sitting `getTracingState` takes a list of variables: `std::shared_ptr<TracingState> getTracingState(const variable_list& vars)`. Shouldn't we instead be using something like flatten rather than `_first_var` here?

Reply:
As for the signature, in `torch/jit/__init__.py` I wrapped `torch._C.get_tracing_state`, so that it's accessible as `torch.jit.get_tracing_state`. I chose to expand the `*args` from the former in the signature of the latter, since this is an operation that the user would have to do frequently when pushing and popping.

As to `flatten` vs `_first_var`: `get_tracing_state` needs at least one variable, so I just stop at the first I find. Is there an advantage with using more than one? Maybe we should check that the variable we encounter is traced at all?

Reply:
Oh, I missed the `*args`!

The advantage to doing more than one is (1): some variables may not be traced (this occurs if someone creates a Variable during the course of tracing), but as long as *one* variable is traced, the expression participates in the trace, (2) we get to do more sanity checking in this case (because we can check that the vars are consistent.)

Reply:
Got it, changing to flatten